test/integration/test_integration.py

```diff
@@ -1197,6 +1197,43 @@ def test_autoquant_compile(self, device, dtype, m1, m2, k, n):
         sqnr = SQNR(out, out2)
         self.assertTrue(sqnr >= 30)
 
+    @parameterized.expand(COMMON_DEVICE_DTYPE)
+    @unittest.skipIf(not TORCH_VERSION_AFTER_2_3, "autoquant requires 2.3+.")
+    def test_autoquant_manual(self, device, dtype):
+        if device != "cuda" and dtype != torch.bfloat16:
+            self.skipTest(f"autoquant currently does not support {device}")
+        if device != "cuda" or not torch.cuda.is_available():
+            self.skipTest(f"autoquant currently does not support {device}")
+        if torch.cuda.is_available() and torch.cuda.get_device_capability() < (8, 0):
+            if dtype == torch.bfloat16:
+                self.skipTest(f"bfloat16 requires sm80+")
+        m1, m2, k, n = 16, 32, 128, 128
+        model = torch.nn.Sequential(
+            torch.nn.ReLU(),
+            torch.nn.Linear(k,n),
+            torch.nn.ReLU(),
+        ).to(device).to(dtype)
+        example_input = torch.randn(m1, k, device=device, dtype=dtype)
+        example_input2 = torch.randn(m2, k, device=device, dtype=dtype)
+        out = model(example_input)
+
+        mod = torchao.autoquant(torch.compile(model), manual_do_autoquant=True)
+        mod(example_input)
+        mod(example_input2)
+        mod.do_autoquant()
+        out2 = mod(example_input)
+        sqnr = SQNR(out, out2)
+        self.assertTrue(sqnr >= 30)
+
+        mod2 = torchao.autoquant(model, manual_do_autoquant=True)
+        mod2(example_input)
+        mod2(example_input2)
+        mod2.do_autoquant()
```

Comment:
the api names could use some work I'd expect `auto_quant` to quantize but it seems to be closer to `prepare_for_auto_quan`t and at least from this test also requires a warmup

Then there's `manual_do_autoquant` but you also have to to call `do_autoquant()`

Reply:
its the same design as torch.compile, which doesn't actually do any compilation, it sets the model up in a way that when it gets run, it will be torch.compiled. This is the same.

If you have multiple inputs, you need a way to prevent that automatic autoquantization from happening after a model run, is there a better name that makes more sense?

Reply:
I guess yeah I'd expect `manual_do_autoquant` to be False for most users, if this is just meant to be a test utility that's a different matter

In case this is necessary then some better names could be

`mod2 = torchao.autoquant(model, quantize_on_inference=False)`

And then `mod2.quantize()`

Reply:
Steps are
1. Prepare a model for autoquantization
2. Defer quantization until all the relevant input shapes are seen
3. Once that list of shapes is seen then perform the autoquantization